namer/namer.cc

```diff
@@ -581,11 +581,11 @@ class SymbolFinder {
             fillAssign(ctx, asgn);
         } else if (!send->recv.isSelfReference()) {
             handleAssignment(ctx, asgn);
+        } else if (!ast::isa_tree<ast::EmptyTree>(lhs->scope)) {
+            handleAssignment(ctx, asgn);
```

Comment:
This hunk was surprising to me; it makes sense, but I also didn't see tests for non-empty scopes and `type_member`/`type_template` definitions.  Did we already have some?

Ah, I see you filed #6416.  👍